daemon/cmd/endpoint.go

```diff
@@ -559,6 +590,33 @@ func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, e
 	return ep, 0, nil
 }
 
+func (d *Daemon) handleOutdatedPodInformer(
+	ctx context.Context,
+	ep *endpoint.Endpoint,
+) (pod *slim_corev1.Pod, k8sMetadata *endpoint.K8sMetadata, err error) {
+	var once sync.Once
+
+	// Average attempt every 10ms.
+	err = resiliency.Retry(ctx, 2*time.Second, 12, func(_ context.Context, _ int) (bool, error) {
+		pod, k8sMetadata, err = d.fetchK8sMetadataForEndpoint(ep.K8sNamespace, ep.K8sPodName)
+		if pod != nil && ep.K8sUID != string(pod.GetUID()) {
```

Comment:
One minor suggestion: handle the case where the CNI-supplied uid is missing. Just in case.

otherwise LGTM.